net/listener_split.go

```diff
@@ -0,0 +1,110 @@
+package net
+
+import (
+	"context"
+	"net"
+	"net/http"
+
+	"github.com/drand/drand/protobuf/drand"
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/soheilhy/cmux"
+	"google.golang.org/grpc"
+)
+
+// NewGRPCListenerForPublicAndProtocol creates a new listener for the Public and Protocol APIs over GRPC with no TLS.
+func NewGRPCListenerForPublicAndProtocol(addr string, s Service, opts ...grpc.ServerOption) Listener {
+	l, err := net.Listen("tcp", addr)
+	if err != nil {
+		panic("tcp listener: " + err.Error())
```

Comment:
Should this return an error instead of panicing?

Reply:
Fixed.